sessions/2019/january/Session_22.md

```diff
@@ -0,0 +1,135 @@
+# SOFTWARE ENGINEERING IN JAVA
+
+## Session 22 (15/01/2019)
+
+- Java Memory Handling
+- Fixing Logger implmeentation
+- Sprint workflow in GitHub project for estimate effort
+- Lint checker
+- Microservices resources
+- Git advices
+
+### Notes
+
+#### Java memory
+
+![Java memory](https://cdncontribute.geeksforgeeks.org/wp-content/uploads/jvm-3.jpg "Java memory")
+
+Java uses their JVM to distribute its own memory use.
+
+In `Heap` is the zone where variables are. Pointers are in `JVM Language Stacks`.
+`PC Registers` is where CPU instructions are stored. `Native Method Stack` is where the methods that Java has built in are, and `Method Area` is self descriptive.
+
+`Class Loader` is where the compiled classes are found when the program is executed. As an example of this, I could to compile a program, and after it, proceed to remove the `.class` file from the `Class Loader` corresponding folder, after that, if I try to run the program will not do it, because the code isn't where is supossed to be, but if I compile again the code, this will not fail and work since the class is recreated into the `Class Loader` zone.
+
+#### Fixing Logger implementation
+
+Several fixes were made to the DefaultLogger class:
```

Comment:
`s/fixes/improvements/` It wasn't really broken :)

Reply:
Funny way to say it!